context.go

```diff
@@ -172,11 +165,15 @@ func unwrapWafResult(ret wafReturnCode, result *wafResult) (res Result, err erro
 
 // Close calls handle.closeContext which calls ddwaf_context_destroy and maybe also close the handle if it in termination state.
 func (context *Context) Close() {
-	defer context.handle.closeContext(context)
-	// Keep the Go pointer references until the end of the context
-	keepAlive(context.cgoRefs)
-	// The context is no longer used so we can try releasing the Go pointer references asap by nulling them
-	context.cgoRefs = cgoRefPool{}
+	context.mutex.Lock()
+	defer context.mutex.Unlock()
```

Comment:
So this is where the mutex lock was forgotten indeed. Very good catch.

Note: having concurrent calls to wafCtx.Run and wafCtx.Close doesn't happen in the current way we instrument. But it's definitely worth making it correct.